### PR TITLE
added section for registry base images

### PR DESCRIPTION
**Pull request by jc-berger**
This PR was made based off the changes Shoubhik and Scott made in this PR: https://github.com/redhat-appstudio/docs.appstudio.io/pull/121

### CODE REVIEW

File: docs/modules/ROOT/pages/how-to-guides/Import-code/proc_importing_code.adoc [hunk 1]
```diff
@@ -214,7 +214,38 @@ For more information about devfile contents requirements, see the "What is outer
...
+
+Retrieve the Red Hat Container Registry token and configure your application to use this token. Using this token improves the build process of your application. 
+
+.Procedure
+
+Retrieve the Red Hat Container Registry token by following these steps:
+
+. Go to link:https://access.redhat.com/terms-based-registry/#/[Registry service account]. 
+. Create a new Registry service account by selecting *New Service Account*.  
+. Retrieve the Registry service account token as a “Docker Configuration”:
```
Comment (jseseCCS) [inline docs/modules/ROOT/pages/how-to-guides/Import-code/proc_importing_code.adoc:227]: lines 227-236: I clicked **New Service Account** and now i'm on the actual **Create a New Registry Service Account** page, and it's not clear to me where a user would do step 3. I see **Name** and **Description** fields ...

File: docs/modules/ROOT/pages/how-to-guides/Import-code/proc_importing_code.adoc [hunk 2]
```diff
@@ -214,7 +214,38 @@ For more information about devfile contents requirements, see the "What is outer
...
+```
+   {
+    "auths": {
+        "registry.redhat.io": {
+        "auth": "MTEwNzAxOTl8-redacted-c2Jvc2U6ZZlNDQ="
+        }
+    }
+    }
+``` 
+.Procedure
```
Comment (jseseCCS) [inline docs/modules/ROOT/pages/how-to-guides/Import-code/proc_importing_code.adoc:237]: i think this needs its own heading. without it, i'm not sure what the procedure is gonna tell me, and i also didn't expect to find a second proc

File: docs/modules/ROOT/pages/how-to-guides/Import-code/proc_importing_code.adoc [hunk 3]
```diff
@@ -214,7 +214,38 @@ For more information about devfile contents requirements, see the "What is outer
...
+    "auths": {
+        "registry.redhat.io": {
+        "auth": "MTEwNzAxOTl8-redacted-c2Jvc2U6ZZlNDQ="
+        }
+    }
+    }
+``` 
+.Procedure 
+
+Configure your application to use the Red Hat Container Registry token by following these steps:
```
Comment (jseseCCS) [inline docs/modules/ROOT/pages/how-to-guides/Import-code/proc_importing_code.adoc:239]: ah, ok. i see. so i'd call procedure 1 something like "Retrieving the ... token from _________," and this second one something like "Configuring your application to ...". then the title of the whole section could be "Retrieving and configuring ...". i'm still not clear, though, where i'm retrieving it from or why.

File: docs/modules/ROOT/pages/how-to-guides/Import-code/proc_importing_code.adoc [hunk 4]
```diff
@@ -214,7 +214,38 @@ For more information about devfile contents requirements, see the "What is outer
...
+        "auth": "MTEwNzAxOTl8-redacted-c2Jvc2U6ZZlNDQ="
+        }
+    }
+    }
+``` 
+.Procedure 
+
+Configure your application to use the Red Hat Container Registry token by following these steps: 
+
+. Go to *Component Settings*.
```
Comment (jseseCCS) [inline docs/modules/ROOT/pages/how-to-guides/Import-code/proc_importing_code.adoc:241]: orient me: where am i now? i've left **Registry service account**, i presume, and am now in RHTAP itself? and where in RHTAP do i find **Component settings**?

File: docs/modules/ROOT/pages/how-to-guides/Import-code/proc_importing_code.adoc [hunk 5]
```diff
@@ -214,7 +214,38 @@ For more information about devfile contents requirements, see the "What is outer
...
+    }
+``` 
+.Procedure 
+
+Configure your application to use the Red Hat Container Registry token by following these steps: 
+
+. Go to *Component Settings*. 
+. Select *Create new build secret*.
+. In the *Create new build secret* modal, add the registry.redhat.io token to be used as a pull secret:
+.. In the *Key* input field, enter *.dockerconfigjson*.
```
Comment (jseseCCS) [inline docs/modules/ROOT/pages/how-to-guides/Import-code/proc_importing_code.adoc:244]: --> In the **Key** field, enter **`.dockerconfigjson`**.

File: docs/modules/ROOT/pages/how-to-guides/Import-code/proc_importing_code.adoc [hunk 6]
```diff
@@ -214,7 +214,36 @@ For more information about devfile contents requirements, see the "What is outer
...
\ No newline at end of file
+=== Building with Red Hat Container Registry tokens
+
+To improve your application-building experience, configure a Red Hat Container Registry token.  
+
+.Procedure
+
+Create the Red Hat Container Registry token by following these steps:
+
+. Go to link:https://access.redhat.com/terms-based-registry/#/[registry service account].
```
Comment (jseseCCS) [inline docs/modules/ROOT/pages/how-to-guides/Import-code/proc_importing_code.adoc:225]: --> Registry Service Accounts [note _s_ on end, too]

Screenshot from 2023-05-26 11-05-33

File: docs/modules/ROOT/pages/how-to-guides/Import-code/proc_importing_code.adoc [hunk 7]
```diff
@@ -214,7 +214,36 @@ For more information about devfile contents requirements, see the "What is outer
...
+
+To improve your application-building experience, configure a Red Hat Container Registry token.  
+
+.Procedure
+
+Create the Red Hat Container Registry token by following these steps:
+
+. Go to link:https://access.redhat.com/terms-based-registry/#/[registry service account]. 
+. Create a new registry service account by clicking *New Service Account*. 
+. Complete the `Name` and `Description` field.
```
Comment (jseseCCS) [inline docs/modules/ROOT/pages/how-to-guides/Import-code/proc_importing_code.adoc:227]: --> Fill in ... 

/? (very small nit. totes take or leave!)

File: docs/modules/ROOT/pages/how-to-guides/Import-code/proc_importing_code.adoc [hunk 8]
```diff
@@ -214,7 +214,36 @@ For more information about devfile contents requirements, see the "What is outer
...
+To improve your application-building experience, configure a Red Hat Container Registry token.  
+
+.Procedure
+
+Create the Red Hat Container Registry token by following these steps:
+
+. Go to link:https://access.redhat.com/terms-based-registry/#/[registry service account]. 
+. Create a new registry service account by clicking *New Service Account*. 
+. Complete the `Name` and `Description` field. 
+. For `Token Information`, click `Docker Configuration`.
```
Comment (jseseCCS) [inline docs/modules/ROOT/pages/how-to-guides/Import-code/proc_importing_code.adoc:228]: should there be a step before this one (new step 4) that says "Click **Create**."?

**Review by jseseCCS — APPROVED** [review]
A few more suggestions/questions, but I trust you and your expertise, so after you address as many as you can, LGTM!

**Review by scoheb — COMMENTED** [review]
LGTM!

File: docs/modules/ROOT/pages/how-to-guides/Import-code/proc_importing_code.adoc (hunk 4 above)
Comment (sbose78) [reply docs/modules/ROOT/pages/how-to-guides/Import-code/proc_importing_code.adoc:247]: Fair ask 👍

File: docs/modules/ROOT/pages/how-to-guides/Import-code/proc_importing_code.adoc [hunk 9]
```diff
@@ -214,7 +214,40 @@ For more information about devfile contents requirements, see the "What is outer
...
 **Solution:** Select a predefined runtime type from the **Review and configure for deployment** view. Make sure you meet all of the {ProductName} <<repository requirements>> for the runtime you select, or configure your own <<custom build and deployment instructions>>.
 
-[role="_additional-resources"]
-.Additional resources
-For information about importing and configuring code to {ProductName} using the CLI, see link:https://redhat-appstudio.github.io/docs.stonesoup.io/Documentation/main/getting-started/getting_started_in_cli/#getting-started-in-the-cli[Getting started in the CLI
-].
\ No newline at end of file
+=== Creating a Red Hat Container Registry token
+
+To improve your application-building experience, configure a Red Hat Container Registry token.
```
Comment (sbose78) [inline docs/modules/ROOT/pages/how-to-guides/Import-code/proc_importing_code.adoc:219]: Wonder if we wish to add a note on why a user might want to use the Red Hat Container Registry token ? 

Example, being able to use registry.redhat.io images as a base image in the Dockerfile - 

```
FROM registry.redhat.io/jboss-eap-7/eap74-openjdk11-openshift-rhel8
```

Comment (jseseCCS) [reply docs/modules/ROOT/pages/how-to-guides/Import-code/proc_importing_code.adoc:219]: totally agree, @sbose78

File: docs/modules/ROOT/pages/how-to-guides/Import-code/proc_importing_code.adoc [hunk 10]
```diff
@@ -214,7 +214,40 @@ For more information about devfile contents requirements, see the "What is outer
...
+
+.Procedure  
+
+. Go to *Component Settings*. 
+. Select *Create new build secret*.
+. In the *Create new build secret* modal, add the *registry.redhat.io* token that you want to use as a pull secret:
+.. In the *Key* field, enter *.dockerconfigjson*.
+.. In the *Select or Enter Name* field, enter *registry-redhat.io*. 
+.. Enter the Registry Service account token and click *Create*.
+.. From your  command line, run the `oc secrets link appstudio-pipeline registry-redhat-io` command.
```
Comment (jseseCCS) [inline docs/modules/ROOT/pages/how-to-guides/Import-code/proc_importing_code.adoc:253]: command needs bold too

**Review by jseseCCS — APPROVED** [review]
@jc-berger 1-2 teeny comments; otherwise, LGTM. BUT, ASAP, we really, _really_ need an explanation of the benefits of these procedures/registry tokens. also, since we've joined this content with the importing and configging content, the relationship between those procedures and these tokens should be evident. (rn, it's not.)

**Review by sbose78 — APPROVED** [review]
Looks good.

File: docs/modules/ROOT/pages/how-to-guides/Import-code/proc_importing_code.adoc [hunk 11]
```diff
@@ -214,7 +214,40 @@ For more information about devfile contents requirements, see the "What is outer
 +
 **Solution:** Select a predefined runtime type from the **Review and configure for deployment** view. Make sure you meet all of the {ProductName} <<repository requirements>> for the runtime you select, or configure your own <<custom build and deployment instructions>>.
 
-[role="_additional-resources"]
-.Additional resources
```
Comment (jseseCCS) [inline docs/modules/ROOT/pages/how-to-guides/Import-code/proc_importing_code.adoc:218]: @jc-berger don't throttle me lol, but [Burr just asked Chloe yesterday](https://redhat-internal.slack.com/archives/C04PZ7H0VA8/p1685542679008949) to add this content back in ASAP. including it in Add'l resources is obv not a make-or-break blocker, but at some point, check w/Chloe and see whether this CLI content is indeed available now to link to. FYI

File: docs/modules/ROOT/pages/how-to-guides/Import-code/proc_importing_code.adoc [hunk 12]
```diff
@@ -214,7 +214,40 @@ For more information about devfile contents requirements, see the "What is outer
...
-For information about importing and configuring code to {ProductName} using the CLI, see link:https://redhat-appstudio.github.io/docs.stonesoup.io/Documentation/main/getting-started/getting_started_in_cli/#getting-started-in-the-cli[Getting started in the CLI
-].
\ No newline at end of file
+=== Creating a Red Hat Container Registry token
+
+After importing your code, configure a Red Hat Container Registry token to improve your application-building experience. You can access this token by creating Red Hat registry service accounts. The registry service accounts enable you to retrieve content from registry.redhat.io. Registry.redhat.io helps you manage the automation of your applications.
```
Comment (jseseCCS) [inline docs/modules/ROOT/pages/how-to-guides/Import-code/proc_importing_code.adoc:219]: SR--> After importing your code, configure a Red Hat Container Registry token to improve your application-building experience. You can access this token by creating a Red Hat registry service account where you can retrieve content from registry.redhat.io to manage the automation of your applications. 

is _registry.redhat.io_ a web address? if so, make it a link?

File: docs/modules/ROOT/pages/how-to-guides/Import-code/proc_importing_code.adoc [hunk 13]
```diff
@@ -214,7 +214,40 @@ For more information about devfile contents requirements, see the "What is outer
...
+        "registry.redhat.io": {
+        "auth": "MTEwNzAxOTl8-redacted-c2Jvc2U6ZZlNDQ="
+        }
+    }
+    }
+``` 
+
+=== Configuring your application to use a Red Hat Container Registry token 
+
+After creating the Red Hat Container Registry token, include the token in your application. When you application includes the Red Hat Container Registry token, you can access registry.redhat.io which improves your automation.
```
Comment (jseseCCS) [inline docs/modules/ROOT/pages/how-to-guides/Import-code/proc_importing_code.adoc:243]: improves automation how?

File: docs/modules/ROOT/pages/how-to-guides/Import-code/proc_importing_code.adoc [hunk 14]
```diff
@@ -214,7 +214,40 @@ For more information about devfile contents requirements, see the "What is outer
...
+
+=== Configuring your application to use a Red Hat Container Registry token 
+
+After creating the Red Hat Container Registry token, include the token in your application. When you application includes the Red Hat Container Registry token, you can access registry.redhat.io which improves your automation. 
+
+.Procedure  
+
+. Go to *Component Settings*. 
+. Select *Create new build secret*.
+. In the *Create new build secret* modal, add the *registry.redhat.io* token that you want to use as a pull secret:
```
Comment (jseseCCS) [inline docs/modules/ROOT/pages/how-to-guides/Import-code/proc_importing_code.adoc:249]: format _registry.redhat.io_ the same everywhere

**Review by jseseCCS — APPROVED** [review]
one typo

**Review by jseseCCS — APPROVED** [review]

**Pull request merged**